Fix intermittent problems on Windows in the XFA search integration test

timvandermeij opened the pull request:
The current test fails intermittently only on Windows for unknown reasons: the code is correct and on Linux it always passes. However, we have already spent quite a lot of time on this test, so rather than spending even more time on it I figured we should look at what behavior the test is trying to check and find an alternative way to do it that can't trigger this intermittent issue anymore.

This commit changes the test to use a term that only exists once in the entire document so we cannot accidentally highlight another match anymore. This doesn't change anything about the behavior that this test aims to check: we still test searching in the XFA layer, we still test that the original term is matched case-insensitively and we still test that that match is actually highlighted. Note that the only objective of the test is confirming that the search functionality covers the XFA layer, so the exact phrase/match is not the interesting bit.

Fixes #17096.

timvandermeij committed:
Fix intermittent problems on Windows in the XFA search integration test

The current test fails intermittently only on Windows for unknown
reasons: the code is correct and on Linux it always passes. However, we
have already spent quite a lot of time on this test, so rather than
spending even more time on it I figured we should look at what behavior
the test is trying to check and find an alternative way to do it that
can't trigger this intermittent issue anymore.

This commit changes the test to use a term that only exists once in the
entire document so we cannot accidentally highlight another match
anymore. This doesn't change anything about the behavior that this test
aims to check: we still test searching in the XFA layer, we still test
that the original term is matched case-insensitively and we still test
that that match is actually highlighted. Note that the only objective of
the test is confirming that the search functionality covers the XFA
layer, so the exact phrase/match is not the interesting bit.

timvandermeij commented:
/botio integrationtest

timvandermeij commented:
/botio integrationtest

calixteman reviewed (approved):
LGTM. Thank you.

calixteman commented:
I added:
https://github.com/mozilla/pdf.js/blob/a9a1195582455bb00a4016656785432b1808d8fc/test/integration/test_utils.mjs#L58
few weeks ago to fix an issue we had we the pdf in the test was reopened in an other test run.

timvandermeij commented:
> I added:
> 
> https://github.com/mozilla/pdf.js/blob/a9a1195582455bb00a4016656785432b1808d8fc/test/integration/test_utils.mjs#L58
> 
> few weeks ago to fix an issue we had we the pdf in the test was reopened in an other test run.

Ah, that explains why my previous hypothesis of e.g. viewer history being kept around between test runs didn't turn out to be the case. Really great that you already thought about and implemented that!
